methods/local_search/nearest_neighbour_flexible_greedy_intraedge.go

```diff
@@ -0,0 +1,31 @@
+package local_search
+
+import (
+	"evolutionary_computation/methods"
+)
+
+func NearestNeighbourFlexibleGreedyIntraEdge(distanceMatrix [][]int, startNode int) []int {
+	// Run random function to get the initial solution
+	selectedIDs := methods.NearestNeighborFlexible(distanceMatrix, startNode)
+	visted := make(map[int]bool)
+	// Make visited map
```

Comment:
Each function that follows the same idea as this one could be stored in one file - `local_search.go` that is in methods package. 
It would import from local_search package for all of the functions as suggested in previous comments